### ISSUE TITLE
Bug: More issues with `event_filter`

### ISSUE THREAD

**Issue by ianhi** [pymmcore-plus/napari-micromanager#261]
#254 doesn't seem to have fully fixed things for me yet unfortunately.

When I launch napari-micromanager from a script:

```python
import napari
from pymmcore_plus import CMMCorePlus

v = napari.Viewer()
dw, main_window = v.window.add_plugin_dock_widget("napari-micromanager")

core = CMMCorePlus.instance()
core.loadSystemConfiguration()

napari.run()
```

then it freezes multiple times as everything is loaded. 

![image](https://github.com/pymmcore-plus/napari-micromanager/assets/10111092/2753818a-3bc5-4fe3-bebd-9010bd007273)

This seems to be due to the event filter as when I comment out this line:
https://github.com/pymmcore-plus/napari-micromanager/blob/f3cb1693a0780df9c02111eee7621b7bf22837f6/src/napari_micromanager/_gui_objects/_toolbar.py#L109

the behavior no longer occurs. I also added some counting code and found that the event filter seems to be called upwards of 20000 times before I even interact with the window.

I've also found that if then go and try to move the window around i get a `Segmentation fault (core dumped)`

```
napari: 0.4.17
Platform: Linux-5.17.15-76051715-generic-x86_64-with-glibc2.35
System: Pop!_OS 22.04 LTS
Python: 3.9.16 | packaged by conda-forge | (main, Feb 1 2023, 21:39:03) [GCC 11.3.0]
Qt: 5.15.2
PyQt5: 5.15.9
NumPy: 1.24.3
SciPy: 1.9.1
Dask: 2023.4.1
VisPy: 0.11.0
magicgui: 0.7.2
superqt: unknown
in-n-out: 0.1.7
app-model: 0.1.4
npe2: 0.7.0

OpenGL:
- GL version: 4.6 (Compatibility Profile) Mesa 22.3.5
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1920x1080, scale 1.0
```

**Comment by ianhi** [comment]
accidentally hit enter before I meant to post so many edits above - but @tlambert03 do you also see this and/or have any insight into potentially lessening the impact of the event filter

**Comment by tlambert03** [comment]
we should ask @fdrgsp ... he's been using it much more than I have.

**Comment by ianhi** [comment]
per a tip from @fdrgsp on zulip using `pyside2` seems to fix it. Though ideally it should work with PyQt5 also :/

**Comment by tlambert03** [comment]
ok will have a look

**Comment by tlambert03** [comment]
I've been unable to reproduce this on macOS 13 with
```
napari: 0.4.17
Platform: macOS-13.3.1-x86_64-i386-64bit
System: MacOS 13.3.1
Python: 3.10.11 | packaged by conda-forge | (main, May 10 2023, 19:07:22) [Clang 14.0.6 ]
Qt: 5.15.2
PyQt5: 5.15.9
```

the event filter is only there to force the toolbar to be on the top.  So, if you want to add an environment variable to disable it, or anything else you want to do to make it work better for you, it's fine with me.

The event filter being called a gazillion times isn't in itself too surprising (any and all events pass like wildfire through the widget tree...).  But it should almost never hit that conditional: `event.type() == QEvent.Type.Move and obj is self`...  (though of course simply evaluating that could perhaps be the problem?  not sure why it would be os specific)

**Comment by tlambert03** [comment]
@ianhi, please recheck.  Since I was unable to reproduce even on PyQt5, I'm inclined to close this, but still waiting to hear from you

**Comment by ianhi** [comment]
I can still reproduce this on python 3.11. It resolves if I `export QT_API=pyside2` before running.

**Comment by ianhi** [comment]
Since I can reproduce I'm also happy to spend a bit of time on this next time I come by HMS

**Comment by tlambert03** [comment]
that would be great too :)

**Comment by ianhi** [comment]
solved by #280
